### ISSUE TITLE
DatePicker: Issue with DatePicker Component when Browser Language is Changed

### ISSUE THREAD

**Issue by SivaiahSankula** [infor-design/enterprise-ng#1774]
**Describe the bug**
Experiencing an issue with the DatePicker component when the browser language is changed. The component behaves unexpectedly in some languages, such as French and English (India), while it works fine in others (like Japanese).

**To Reproduce**
- Go to https://main-enterprise.demo.design.infor.com/components/datepicker/example-range.html?locale=fr-FR
- Selected the dates from 2nd August 2024 to 9th August 2024. 
- Reopened the picker, it incorrectly showed the selected dates as 8th February 2024 to 8th September 2024.
- Selected the dates from 1st August 2024 to 31st August 2024. Upon reopening the picker, none of the previously selected dates were highlighted as selected.
- Observe that the selected date range is different from the originally selected date.

**Expected behavior**
Selected date range is different, it should not be.

**Version**
- ids-enterprise-ng: 18.2.1

**Additional context**
Please refer below Stackblitz link 
[DatePicker Example StackBlitz](https://stackblitz.com/edit/ids-quick-start-1821-9stvrp?file=src%2Findex.html)

**Comment by tmcconechy** [comment]
Looked at this briefly bit where in the code do you set the `Soho.Locale` exactly? I would look there. We dont do anything with the browser language so you need to map that to the `Soho.Locale.set` and we dont change values in the field automatically so you might need to do that as well when you change browser language.

In web components it NOW will change the value in the field when changing locale (not browser language. See https://main.wc.design.infor.com/ids-date-picker/example.html and change in the

**Comment by SivaiahSankula** [comment]
Hi @tmcconechy , In our project we are setting the browser language, using Soho.Locale.set only in soho-locale-initializer.service.ts .
Please find the below screen shot for more understanding. 

<img width="1060" alt="image" src="https://github.com/user-attachments/assets/5735447a-25c5-4c6f-9550-461617523c2a">


**navigator.language** will provide the browser language which we are setting up.

I can refer the example which you have shared that is in web-components. can this functionality or changes will be same for the enterprise-ng, since we are using ids-enterprise-ng in our project?.

**Comment by tmcconechy** [comment]
@SivaiahSankula so you just want when change locale the the value in the field will also update? Maybe we can do this with a setting but i dont think we can make it the default or this would break as its been this way for many years.

**Comment by SivaiahSankula** [comment]
@tmcconechy I think we are getting out of track here. What I simply meant to say that Date picker range is not working as expected in case of certain languages like French as it has a different format compared to other languages. So could you try the date picker with French as locale language. If you check the stackblitz we are setting the language as French in our example. We don't need an enhancement to set the language in DatePicker. We want the date picker range to be working as expected in French language as well.
 
 
 
 Attaching a video for your reference. In the video you can see when we select language as en-Us and use datepicker we select a range from 1st August 2024 to 9th August 2024, if we open the date picker again it shows the same selected  range, if we change the language to French and select the same range we get the range as 8th January 2024 to 8th September 2024, which is deviating from the expected behavior. This is the issue we want you to check
 
 

https://github.com/user-attachments/assets/895e712a-d55d-40ed-9bf0-f1461ed891c9

**Comment by tmcconechy** [comment]
OK good lets focus on that then. @SivaiahSankula 

Can you update the step above so its more clear? For example changing the browser locale is the confusing part because we dont handle that. But in the example you posted if i open the picker a second time after picking it is not showing me the dates? Is that the issue really?

The video is over two minutes long. A lot of clicking and i cant really follow where the bug is so looking for a simpler report.

Maybe we can try the examples if need:
https://main-enterprise.demo.design.infor.com/components/datepicker/example-range.html
https://main-enterprise.demo.design.infor.com/components/datepicker/example-range.html?locale=fr-FR

**Comment by SivaiahSankula** [comment]
Hello @tmcconechy .
Regarding your **question**: "_But in the example you posted, if I open the picker a second time after selecting a date, it does not show me the dates? Is that really the issue?_"

**Answer**: Exactly, that's the issue we're facing. This problem occurs only in certain languages, such as **French**. In other languages, it works as expected.

To help clarify, I'm uploading another video using the **demo example you provided**.
https://main-enterprise.demo.design.infor.com/components/datepicker/example-range.html?locale=fr-FR

Sample Cases:

- Case 1: I selected the dates from `2nd August 2024 to 9th August 2024`. When I reopened the picker, it incorrectly showed the selected dates as `8th February 2024 to 8th September 2024`.
- Case 2: I selected the dates from `1st August 2024 to 31st August 2024`. Upon reopening the picker, none of the previously selected dates were highlighted as selected.

Additionally, I tried changing the locale value in the URL to en-US (https://main-enterprise.demo.design.infor.com/components/datepicker/example-range.html?locale=en-US), and in this configuration, the above cases work as expected. This suggests the issue might be related to the date format in certain locales.



https://github.com/user-attachments/assets/f4958338-527c-4a66-9ece-69a128cbae4e
